core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -483,7 +483,7 @@ class KafkaApis(val requestChannel: RequestChannel,
           case (topicPartition, partitionData) =>
             try {
               if (partitionData.committedMetadata() != null
-                && partitionData.committedMetadata().length > config.offsetMetadataMaxSize)
+                && partitionData.committedMetadata().lengthCompare(config.offsetMetadataMaxSize) > 0)
```

Comment:
`committedMetadata` returns a `String`, so this change doesn't seem like an improvement. Similar comment for other similar changes where the underlying collection is _not_ a Scala List.

Reply:
Good point. The updated PR only contains entries where we are comparing `Seq` or `Iterables`.